app/lib/contentdm/collection.rb

```diff
@@ -0,0 +1,31 @@
+###
+# This class creates and returns a collection object
+# It will first look for a Collection with the title it is given
+# if it finds one, it uses that collection. If it doesn't find one
+# it will create a new collection with the given title.
+module Contentdm
+  class Collection
+    ##
+    # @param title [String]
+    def initialize(title)
+      @title = title
+      @collection_id
+    end
+
+    ##
+    # @return [ActiveFedora::Base]
+    def find_or_create
+      ::Collection.find_each(title: @title) do |collection|
+        @collection_id = collection.id
+      end
```

Comment:
It might be slightly confusing to set the `@collection_id` inside this loop.  We expect `find_each` to only find one collection, but it's possible that it would find more than one, and then the `@collection_id` would get set to the last collection in the loop.  Without explaining why we're doing that, someone might read this code in the future and think it's a mistake.

I think it would be clearer if you made it explicit that we're only working with one collection.  You could do something like this:
```
collection = Collection.where(title_tesim: @title).first || ::Collection.new(@title)
collection.visibility = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC
collection.save!
@collection_id = collection.id
```